File: WebPDAM/src/app/controller/services/schedule.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

import { ResponseContainer } from '../interfaces/response-container';
import { ScheduleResponse } from '../interfaces/schedule-response';
import { Schedule } from '../model/schedule';
import { AuthService } from './auth.service';

const baseUrl = `${environment.apiUrl}/schedules`;

@Injectable({
  providedIn: 'root'
})
export class ScheduleService {

  constructor(private http: HttpClient, private authService: AuthService) { }

  requestOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json',
      Authorization: `Bearer ${this.authService.getToken()}`
    })
  };

  create(schedule: Schedule): Observable<Schedule> {
    return this.http.post<Schedule>(`${baseUrl}`, schedule, this.requestOptions);
  }

  createMany(schedules: Schedule[]): Observable<Schedule> {
    return this.http.post<Schedule>(`${baseUrl}/many`, schedules, this.requestOptions);
  }

  getAll(): Observable<ResponseContainer<ScheduleResponse>> {
    return this.http.get<ResponseContainer<ScheduleResponse>>(`${baseUrl}`, this.requestOptions);
  }

  getAllFiltered(query: string): Observable<ResponseContainer<ScheduleResponse>> {
    return this.http.get<ResponseContainer<ScheduleResponse>>(`${baseUrl}?${query}`, this.requestOptions);
  }

  getOneDayFiltered(query: string): Observable<ScheduleResponse[]> {
    return this.http.get<ScheduleResponse[]>(`${baseUrl}/oneday?${query}`, this.requestOptions);
  }

  edit(id: string, data: Schedule) {
    return this.http.put<Schedule>(`${baseUrl}/${id}`, data, this.requestOptions);
  }

  delete(id: string) {
    return this.http.delete(`${baseUrl}/${id}`, this.requestOptions);
  }
}

File: AndroidPDAM/app/src/main/java/com/palma/pdam/retrofit/service/ScheduleService.java
package com.palma.pdam.retrofit.service;

import com.palma.pdam.data.response.ResponseContainer;
import com.palma.pdam.data.response.ScheduleResponse;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

public interface ScheduleService {
    String BASE_URL = "/schedules";

    @GET(BASE_URL + "/daily")
    Call<ResponseContainer<ScheduleResponse>> getDaily(@QueryMap Map<String, String> options);

    @GET(BASE_URL + "/oneday")
    Call<List<ScheduleResponse>> getOneDay(@QueryMap Map<String, String> options);

}
